src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -143,6 +143,7 @@ public Block[] Process(Hash256 newBranchStateRoot, List<Block> suggestedBlocks,
                 }
 
                 preBlockStateRoot = processedBlock.StateRoot;
+                _stateProvider.Reset(resizeCollections: true);
```

Comment:
this would clear those caches per block? Maybe would be better to clear it per batch?

Reply:
Blocks are fast anyway; builds up a lot of new junk that wasn't there before.

Can look at optimizing for a cross block cache as a different exercise?